### DISCUSSION TITLE
How to get started building apps with `gitoxide`?

### DISCUSSION THREAD

**Discussion by rudrabhoj** [GitoxideLabs/gitoxide#508] [General]
### Summary 💡

Hey I am building a free and open-source alternative to Gitkraken. Trying to get around gitoxide_core library, but I am unable to find any documentation at all. Best I am able to find is a module/function list.

Is any other (more thorough) documentation available? If not, code samples? What would be the best way to get started with Gitoxide?

**Comment by Byron** [comment]
Thanks for your interest! `gitoxide-core` is intentionally undocumented, it's a playground to build the `gix` and `ein` CLI and the code in there will be shuffled around a lot.

What one wants to use for application development is `git-repository`, which is fully documented as well. There are a couple of `./examples/` as well. Depending on your appetite for risk you might want to stick with `git2` for the time-being as it will have all the features you probably need. With `git-repository` and its entry in `crate-status.md` you will probably know whether or not to start with `gitoxide` at first if you have an analysis of the order of features you would like to implement.

**Comment by wmstack** [comment]
One thing I find useful with projects with obscure documentation is to look through `sourcegraph` for how other people utilize the project:

https://sourcegraph.com/search?q=context:global+use+git_repository&patternType=standard

On the right, you can see all the public GitHub repositories that use that crate. Hope that helps.
